js/sketch.js

```diff
@@ -0,0 +1,86 @@
+class Helper {
+    generateRandomNumber(min, max) {
+        return Math.floor(Math.random() * (max - min + 1)) + min;
+    }
+
+    generateRandomColor() {
+        return [
+            this.generateRandomNumber(0, 255),
+            this.generateRandomNumber(0, 255),
+            this.generateRandomNumber(0, 255),
+        ];
+    }
+}
+
+class Circle {
+    x;
+    y;
+    size;
+    stepsX;
+    stepsY;
+
+    constructor(x, y, size, stepsX, stepsY) {
+        this.x = x;
+        this.y = y;
+        this.size = size;
+        this.stepsX = stepsX;
+        this.stepsY = stepsY;
+        this.color = new Helper().generateRandomColor();
+    }
+
+    createCircle() {
+        circle(this.x, this.y, this.size);
+    }
+
+    fillColor() {
+        fill(
+            `rgba(${this.color[0]}, ${this.color[1]}, ${this.color[2]}, 0.5)`
+        );
+    }
+
+    moveCircle() {
+        this.x += this.stepsX;
+        this.y += this.stepsY;
+
+        if (this.x > width || this.x < 0) {
```

Comment:
fix these conditions so that the circle doesnt pass through the end of the screen before bouncing